### PR TITLE
Impostors and Palettes

### LINKED ISSUE #7: Separate palettes into banks which can be independently toggled
It feels kind of weird to have both canon yokers and non-canon yokers both appearing at once. I think it would be cool to separate all of the palettes we have into groups, which can independently be toggled on/off and maybe even made more/less frequent. That way, if someone like me prefers to only have the non-canon yokers, they can. These kind of structural changes will also help us keep palettes organized and less chaotic if and when we make more of them.

### LINKED ISSUE #9: Make impostor frequency a slider
Since the impostors do look pretty neat and there are quite a few of them now, it'd be cool to customize how frequently they show up. I don't have any problems with allowing someone to go anywhere from 0% to 100% impostors, but we need to make sure the UI still "prefers" to keep them rare.

### LINKED ISSUE #12: Make the build artifact called yok.scr
VS builds our project into an executable called `yokscr.exe`, but we really want it to be called `yok.scr`. I have no idea if this is even possible, but I figure it must be.

### PR DESCRIPTION
**Pull request by McAwesome123**

### CODE REVIEW

**Comment by McAwesome123** [comment]
Need to fix the crash from the color count being set too high when changing the palettes option to something other than all.

File: configdialog.cpp
Comment (pierce-smith1) [inline configdialog.cpp:53]: You might be surprised to hear but I have no problem with how this is being done.

File: resources.h
Comment (pierce-smith1) [inline resources.h]: Let's do whatever we can to avoid repeating ourselves here since this is just asking for trouble. Here's an idea: since these palette names are just the numbers from 0 to 47 or whatever, let's just generate it at run time with a function:

```c++
inline std::vector<PaletteName> make_all_palettes() {
    std::vector<PaletteName> all_palettes((size_t) PaletteName::_PALETTE_COUNT);
    std::generate(all_palettes.begin(), all_palettes.end(), [n = 0]() mutable { return (PaletteName) n++; });
    return all_palettes;
}

// ...
{ PaletteGroup::All, make_all_palettes() }
```

This is some of the weirdest fucking c++ I've ever seen but it does seem to work. I stole it from [here](https://en.cppreference.com/w/cpp/algorithm/generate).

File: spritecontrol.cpp
Comment (pierce-smith1) [inline spritecontrol.cpp]: Be sure to clean these comments up.

File: configdialog.cpp
Comment (pierce-smith1) [inline configdialog.cpp]: Something about all of this is hurting my 3 am brain and I don't know what to make of it. I'll come back to this.

**Review by pierce-smith1 — CHANGES_REQUESTED** [review]

File: spritecontrol.cpp
Comment (pierce-smith1) [inline spritecontrol.cpp]: We really want `MaxColors` to be a value between 0.0 and 1.0, but we can't just do that because people's existing configs will be outside of that range. So I guess this is fine for now.

Comment (pierce-smith1) [reply spritecontrol.cpp]: However - styling!! `snake_case` for locals please. I'm pointing this out because VS won't enforce it, even with the new .editorconfig.

File: spritecontrol.cpp
Comment (pierce-smith1) [inline spritecontrol.cpp]: I'm scared of this. If the result of `round(maxColors)` is greater than `bag_of_palettes.size()`, the program will try to erase from an empty list and invoke undefined behavior. We cannot mathematically reason that this will never be the case because of floating-point imprecision. 

Since it's so important for this to not exceed `bag_of_palettes.size()`, I think we should do something like this, even though it is pretty clunky:

```c++
if (round(maxColors) > bag_of_palettes.size()) {
    maxColors = bag_of_palettes.size();
}
```

File: spritecontrol.cpp
Comment (pierce-smith1) [inline spritecontrol.cpp:30]: I may have said it already, but the cubic is pretty clever here.

File: configdialog.cpp
Comment (pierce-smith1) [inline configdialog.cpp]: OK, here are some thoughts. First, I don't think we should show a message box in these cases, we should just silently fix the value and move on. I have a lot to say about this but really it just boils down to "I don't want to annoy users with a message box over something this unimportant."

Second, it feels like this validation + repair stuff really wants to be a separate system in the code. It might be as simple as just having a function `is_config_value_in_range` that we can call rather than doing these if checks manually. I want to think about this some more though, because I'm not 100% sure how we should design it, so I'll make another issue for it and we can address it later.

File: configdialog.cpp
Comment (pierce-smith1) [inline configdialog.cpp]: Why does there need to be a `static_cast` here, but not on the previous if? And as a style point (that VS is not set up to enforce), we should use C style casts instead of `static_cast`.

File: Resource.rc
Comment (pierce-smith1) [inline Resource.rc:1]: image

Two comments on the dialog: 
* The label "Palette" should maybe be pluralized into "Palettes," since we don't want people getting confused and thinking they're selecting one palette for all the yokers.
* The whole "Family" box a just a touch too tall, with a little bit too much empty space on the bottom. It could be shrunk just a tiny bit.

**Review by pierce-smith1 — CHANGES_REQUESTED** [review]
I think this got everything out of my system. I'm going to wait until this PR gets merged before merging any of my other ones, because I want to impending apocalyptic merge conflict shitstorm to fall on me, not you.

File: configdialog.cpp
Comment (McAwesome123) [reply configdialog.cpp]: Wasn't aware that a C-style cast works here. And it's needed because the compiler doesn't like comparing an int to an enum class member.

**Comment by McAwesome123** [comment]
6 am programming goes brrrr

**Review by pierce-smith1 — APPROVED** [review]
It's release time motherfucker.

**Pull request merged**
